### PR TITLE
Handles argument passing to `run_closed_loop` script

### PR DESCRIPTION
**Pull request by eudesf**
#### Introduction

Fixes argument passing to `run_closed_loop` script, so parameters like `--settings-path` can get passed through. 

#### Changes

This change sends all the arguments received on `run_closed_loop` script to the started modules subprocesses so they can be read and handled accordingly.

#### Behavior

The `--settings-path` parameter is being ignored on `run_closed_loop` script.  With this change, that parameter and any other parameter passed will be captured and handled.

### CODE REVIEW

**Review by NewtonSander — CHANGES_REQUESTED** [review]
With the code changes from this PR the arguments are passed to the processes, but with no modifications to the config files we have a crash, as the default `settings.yaml` doesn't contain the content required by the decoder and the center out reach task.

We also miss user documentation mentioning that a user can pass arguments to this script and we also miss some CLI documentation, like the following:
```
encoder --help
usage: encoder [-h] [--settings-path SETTINGS_PATH]

Run encoder.

optional arguments:
  -h, --help            show this help message and exit
  --settings-path SETTINGS_PATH
                        Path to the settings.yaml file.
```

Before proceeding with this changes I would like to discuss if this solution (sending the same arguments to all processes) is better than allowing the user to send specific arguments to each process

**Comment by eudesf** [comment]
> With the code changes from this PR the arguments are passed to the processes, but with no modifications to the config files we have a crash, as the default `settings.yaml` doesn't contain the content required by the decoder and the center out reach task.
> 
> We also miss user documentation mentioning that a user can pass arguments to this script and we also miss some CLI documentation, like the following:
> 
> ```
> encoder --help
> usage: encoder [-h] [--settings-path SETTINGS_PATH]
> 
> Run encoder.
> 
> optional arguments:
>   -h, --help            show this help message and exit
>   --settings-path SETTINGS_PATH
>                         Path to the settings.yaml file.
> ```
> 
> Before proceeding with this changes I would like to discuss if this solution (sending the same arguments to all processes) is better than allowing the user to send specific arguments to each process

@NewtonSander please check again! I've included specific parameters for each process started. :)

File: src/tasks/run_closed_loop.py [hunk 1]
```diff
@@ -16,10 +46,20 @@ def run():
...
-    center_out_reach = subprocess.Popen(["center_out_reach"])
+    args = _parse_args()
+
+    SETTINGS_PATH_PARAM = "--settings-path"
+
+    nds_params = _build_param_from_arg(args, "nds_settings_path", SETTINGS_PATH_PARAM)
+    decoder_params = _build_param_from_arg(
+        args, "decoder_settings_path", SETTINGS_PATH_PARAM
+    )
+    task_params = _build_param_from_arg(args, "task_settings_path", SETTINGS_PATH_PARAM)
```
Comment (NewtonSander) [inline src/tasks/run_closed_loop.py:57]: You don't need the `getattr` inside `_build_param_from_arg`, if the argument is not passed to the script you will have None as value
```suggestion
    nds_params = _build_param_from_arg(args.nds_settings_path, SETTINGS_PATH_PARAM)
    decoder_params = _build_param_from_arg(
        args.decoder_settings_path, SETTINGS_PATH_PARAM
    )
    task_params = _build_param_from_arg(args.task_settings_path, SETTINGS_PATH_PARAM)
```

**Review by NewtonSander — CHANGES_REQUESTED** [review]

File: src/tasks/run_closed_loop.py (hunk 1 above)
Comment (eudesf) [reply src/tasks/run_closed_loop.py]: Nice! I've already sent the update.

**Review by NewtonSander — APPROVED** [review]

**Pull request merged**